core/pom.xml

```diff
@@ -30,6 +30,10 @@
          <artifactId>jgroups</artifactId>
       </dependency>
 
+      <dependency>
```

Comment:
Ok, I created another module called `infinispan-cloud` and moved all cloud related configs there. I also added proper information in upgrading guide.

//cc @alanfx @tristantarrant